### PR TITLE
🐛 Add ability to cache comments count endpoint

### PR DESCRIPTION
**Pull request by lenabaidakova**
closes TryGhost/Team#2094

PR to comments repo https://github.com/TryGhost/comments-ui/pull/12

### CODE REVIEW

File: ghost/core/core/shared/config/defaults.json [hunk 1]
```diff
@@ -140,6 +140,9 @@
         },
         "contentAPI": {
             "maxAge": 0
+        },
+        "comments": {
```
Comment (naz) [inline ghost/core/core/shared/config/defaults.json:144]: I do like the `contentAPI` convention used above. This is an API we are caching, so maybe `commentsAPI` would be a better fit. Also, with the discussion on the issue itself _if_ we end up only caching the comments endpoint long-term, than `commentsCountAPI` would be even better name.

File: ghost/core/core/server/web/comments/routes.js [hunk 2]
```diff
@@ -1,19 +1,22 @@
...
 const {http} = require('@tryghost/api-framework');
+const shared = require('../shared');
 
 const bodyParser = require('body-parser');
 const membersService = require('../../../server/services/members');
 
 module.exports = function apiRoutes() {
     const router = express.Router('comment api');
 
+    router.use(shared.middleware.cacheControl('public', {maxAge: config.get('caching:comments:maxAge')}));
```
Comment (naz) [inline ghost/core/core/server/web/comments/routes.js:13]: The issue we are solving here is for one specific endpoint and not the whole API. I think we would be most safe to apply this caching rule to the counts endpoint first to unblock the issue and do more research around correct approach for the whole Comments API

**Review by naz — CHANGES_REQUESTED** [review]
These changes look good. The controller/api level changes are good to go into main I think. The part around setting a custom cache header should to be endpoint-specific.

File: ghost/core/core/shared/config/defaults.json (hunk 1 above)
Comment (lenabaidakova) [reply ghost/core/core/shared/config/defaults.json]: Renamed it to `commentsCountAPI`

File: ghost/core/core/server/web/comments/routes.js (hunk 2 above)
Comment (lenabaidakova) [reply ghost/core/core/server/web/comments/routes.js]: Added cache just of counts API

**Comment by lenabaidakova** [comment]
@naz , thank you for explanation 🙌 

Fixed next issues:
- Renamed setting in config to `commentsCountAPI`
- Applied just for one count API
- Added `stale-while-revalidate` option

File: ghost/mw-cache-control/lib/mw-cache-control.js [hunk 3]
```diff
@@ -12,10 +12,18 @@ const isString = require('lodash/isString');
  * @param {'public'|'private'} profile Use "private" if you do not want caching
  * @param {object} [options]
  * @param {number} [options.maxAge] The max-age in seconds to use when profile is "public"
+ * @param {number} [options.staleWhileRevalidate] The stale-while-revalidate in seconds to use when profile is "public"
```
Comment (naz) [inline ghost/mw-cache-control/lib/mw-cache-control.js:15]: Oh exciting! Nice 😊

**Review by naz — APPROVED** [review]
Yup! All of this looks about right 👌  If I had to be extra nit-picky about it, would separate it into two commits and `staleWhileRevalidate` functionality there with specific explanation for whys and hows 😊

**Comment by lenabaidakova** [comment]
> Yup! All of this looks about right 👌 If I had to be extra nit-picky about it, would separate it into two commits and `staleWhileRevalidate` functionality there with specific explanation for whys and hows 😊

Fixed :)

**Pull request merged**